<?xml version="1.0"?>
<launch>
  <arg name="transport" default="udp"/>
  <arg name="scanner_ip" default="169.254.253.4"/>
  <arg name="device" default="R2000"/>
  <arg name="start_angle" default="-1800000"/>
  <arg name="max_num_points_scan" default="0"/>

  <include file="$(find pf_description)/launch/pf_bringup.launch" >
    <arg name="scanner" value="r2000" />
  </include>

  <!-- R2000 Driver -->
  <node pkg="pf_driver" type="ros_main" name="r2000_node" output="screen">
    <param name="transport" type="string" value="$(arg transport)"/>
    <param name="scanner_ip" type="string" value="$(arg scanner_ip)"/>
    <param name="port" type="string" value="0"/>
    <param name="frame_id" type="string" value="r2000"/>
    <param name="scan_topic" type="string" value="scan"/>
    <param name="device" type="string" value="$(arg device)"/>
    <param name="start_angle" type="int" value="$(arg start_angle)"/>
    <param name="max_num_points_scan" type="int" value="$(arg max_num_points_scan)"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="static_transform_publisher" args="0 0 0 0 0 0 1 /base_link /scanner 100"/>
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pf_driver)/rviz/r2000.rviz" />
</launch>
